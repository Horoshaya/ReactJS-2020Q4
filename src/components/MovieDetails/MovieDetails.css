.wrapper {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 317px;
  color: #ffffff;
  background: #000000;
}

.header {
  max-width: 1000px;
  text-align: right;
  margin: auto;
  padding: 20px 0;
}

.closeIcon {
  color: #ffffff;
  padding: 15px;
  cursor: pointer;
}

.closeIcon:before {
  content: '\2715';
}

.title {
  font-size: 30px;
}

.vote {
  margin-left: 20px;
  padding: 15px;
  border-radius: 50%;
  color: #498124;
  border: 1px solid #ffffff;
}

.movieHeader {
  display: flex;
}

.movieWrapper {
  display: flex;
  justify-content: space-between;
  max-width: 1000px;
  margin: auto;
}

.movieInfo {
  margin-left: 20px;
}

.posterPath {
  height: 250px;
}
