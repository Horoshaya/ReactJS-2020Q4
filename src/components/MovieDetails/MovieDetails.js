import React from 'react';
import { connect } from 'react-redux';
import { useParams, Link } from 'react-router-dom';
import { getMoviesFromIdThunk } from '../../store/middlewares';

import styles from './MovieDetails.css';

const MovieDetails = (props) => {
  const { id } = useParams();
  props.dispatch(getMoviesFromIdThunk(id));

  const { movie } = props;

  return (
    <Link to="/">
      {movie ? (
        <div className={styles.wrapper}>
          <div className={styles.header}>
            <span className={styles.closeIcon} />
          </div>
          <div className={styles.movieWrapper}>
            <img className={styles.posterPath} src={movie.poster_path} />
            <div className={styles.movieInfo}>
              <div className={styles.movieHeader}>
                <h2 className={styles.title}>{movie.title}</h2>
                <p className={styles.vote}>{movie.vote_average}</p>
              </div>
              <div>
                <p>{movie.overview}</p>
              </div>
            </div>
          </div>
        </div>
      ) : null}
    </Link>
  );
};

const mapStateToProps = (state) => ({
  movie: state.movieDataReducer.movie,
});

export default connect(mapStateToProps)(MovieDetails);
