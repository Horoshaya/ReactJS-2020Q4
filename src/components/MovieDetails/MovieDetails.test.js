import React from 'react';
import rerender from 'react-test-renderer';
import MovieDetails from './MovieDetails';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';

const initialStore = {
  allMoviws: [],
  movieDataReducer: {
    movie: [
      {
        genres: ['Drama', 'Romance'],
      },
    ],
  },
};
const movie = configureStore([])(initialStore);
movie.dispatch = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useParams: jest.fn().mockReturnValue({ id: 'test01' }),
}));

describe('when MovieDetails', () => {
  test('then snapshot created', () => {
    const component = rerender.create(
      <Provider store={movie}>
        <Router>
          <MovieDetails movie={movie} />
        </Router>
      </Provider>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
