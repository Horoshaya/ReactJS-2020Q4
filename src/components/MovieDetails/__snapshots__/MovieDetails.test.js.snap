// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when MovieDetails then snapshot created 1`] = `
<a
  href="/"
  onClick={[Function]}
>
  <div>
    <div>
      <span />
    </div>
    <div>
      <img />
      <div>
        <div>
          <h2 />
          <p />
        </div>
        <div>
          <p />
        </div>
      </div>
    </div>
  </div>
</a>
`;
